### ISSUE TITLE
Incorrect set variable de-duplication

### ISSUE THREAD

**Issue by ttuegel** [kframework/kore#1207]
During substitution normalization, we de-duplicate element variable substitutions in this way:

```
x = t₁ ∧ ... ∧ x = tₙ
x = (t₁ ∧ ... ∧ tₙ)
```

Set variables are de-duplicated in the same way, which is not correct. Set variables should be de-duplicated this way:

```
@x = t₁ ∧ ... ∧ @x = tₙ
@x = t₁ ∧ t₁ = t₂ ∧ ... ∧ t₁ = tₙ
```

These transformations are equivalent for element variables.

See also: https://runtimeverification.slack.com/archives/CC360GUTG/p1572914597064200

**Comment by ttuegel** [comment]
During the iteration planning meeting, we discussed this issue, but could not come to a resolution. We don't actually need to implement set variable deduplication right now because it is not required by any semantics. We should implement an error message in case it is required.

In [`Kore.Step.Simplification.SubstitutionSimplifier.deduplicateSubstitution`](https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/Simplification/SubstitutionSimplifier.hs#L198), the [second argument](https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/Simplification/SubstitutionSimplifier.hs#L215) is the set of substitutions. The keys of that `Map` are the substitution variable, and the `NonEmpty` list of values are all the assignments for that variable. We should add an assertion that if any key is a `SetVar`, then its list of assignments `isSingleton` (`isSingleton = isJust . getSingleton`).

**Comment by ttuegel** [comment]
Fixed in #1224.
